src/lib/Libifl/int_rdrpy.c

```diff
@@ -83,18 +84,22 @@ PBSD_rdrpy_sock(int sock, int *rc)
 		return NULL;
 	}
 
-	DIS_tcp_funcs();
-	old_timeout = pbs_tcp_timeout;
-	if (pbs_tcp_timeout < PBS_DIS_TCP_TIMEOUT_LONG)
-		pbs_tcp_timeout = PBS_DIS_TCP_TIMEOUT_LONG;
+	if (prot == PROT_TCP) {
+		DIS_tcp_funcs();
```

Comment:
dont u have to set it to DIS_tpp_funcs() in case of PROT_TPP? If the function pointers were pointing to TCP in an earlier call - then we would read from the wrong buffers?

Granted, the DIS_xxx_funcs() is the lousiest possible code design! We should change this as soon as we call (perhaps with the flatbuffers RFE)